src/complex/DataStructure/DataStructure.cpp

```diff
@@ -88,6 +88,40 @@ LinkedPath DataStructure::getLinkedPath(const DataPath& path) const
   }
 }
 
+LinkedPath DataStructure::makePath(const DataPath& path)
```

Comment:
Since we want to report errors in the filter parameter, we should change this to a `Result<LinkedPath>` or at the very least `std::optional<LinkedPath>`.

Reply:
Is there an example of doing this in the code?

Reply:
using Result<LinkedPath> just gives a compiler error. Not sure that API exists to be able to do that. Suggest we just merge this PR.

Reply:
```cpp
Result<LinkedPath> DataStructure::makePath(const DataPath& path)
{
  // actually make the path
  if(!sucess)
  {
     return MakeErrorResult<LinkedPath>(-1, "Unable to create path");
  }

  return {linkedPath};
}
```
Something like this?

Also what was the compile error?

And there's always `std::optional<LinkedPath>`.

Reply:
Done

Reply:
Looks like we don't have copy/move assignment for LinkedPath and that's causing the compile issue. Though since we have copy/move constructors I'd say there's no reason we shouldn't.